src/google/cloud/ndb/context.py

```diff
@@ -148,6 +178,8 @@ def use(self):
         try:
             yield self
         finally:
+            if prev_context:
```

Comment:
Why have two caches if we're just going to sync the two?

Reply:
My assumption, from looking at the original implementation, is that it's necessary because futures started before the transaction was initiated can only be allowed to modify the cache context outside of the transaction; they can't be allowed to change the transaction's cache during the transaction itself. However, I'm not fully comfortable with this logic because keys may be inaccurately overwritten after the transaction is complete. For better or for worse, this is faithful to the original implementation: see https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/context.py#L884

Do you think we should stick with this approach, unify the caches so that they are the same object, or make a more complex update procedure that keeps track of touched and untouched keys and only syncs ones that were touched during the transaction?

Reply:
Hmmm. I guess when in doubt, follow the original implementation. Unfortunately the original is complex enough that it's tough to reason through all the potential use/edge cases. You'd like to trust that the first authors did think all those through. Maybe they did!